### PR TITLE
Added PDF 2d-resolution export

### PR DESCRIPTION
**Pull request by tzjtan**
Allow PDF exporting resolution to be specified as a tuple too, to get asymmetric DPIs sometimes used by sticker/inkjet printers, e.g 600x300dpi.

### CODE REVIEW

**Comment by radarhere** [comment]
Can you provide a link to documentation or something similar online that shows that printers use this?

File: src/PIL/PdfImagePlugin.py [hunk 1]
```diff
@@ -211,8 +218,8 @@ def _save(im, fp, filename, save_all=False):
                                     MediaBox=[
                                         0,
                                         0,
-                                        int(width * 72.0 / resolution),
-                                        int(height * 72.0 / resolution)
+                                        int(width * 72.0 / pdf_width_res),
+                                        int(height * 72.0 / pdf_height_res)
```
Comment (hugovk) [inline src/PIL/PdfImagePlugin.py]: Can these values ever be zero?

Comment (tzjtan) [reply src/PIL/PdfImagePlugin.py]: No, these PDF resolutions should always be positive. Are there any changes required?

Comment (tzjtan) [reply src/PIL/PdfImagePlugin.py]: > 
> 
> Can you provide a link to documentation or something similar online that shows that printers use this?

I surveyed the top 2/3 lasers, inkjet and label printers from the US Amazon top seller list (3 Apr 2019) and find the following print resolutions:
Inkjet no. 1 seller - HP OfficeJet 3830 AIO Printer: 4800 x 1200dpi
Inkjet no. 2 seller - HP ENVY 5055: 1200dpi (symmetric)
Inkjet no. 3 seller - Canon PIXMA TS5120: 4800 x 1200dpi
Laser no. 1 seller - Brother monochrome HL-L2350DW: 1200dpi (symmetric)
Laser no. 2 seller - Brother MFC-L2710dw: 2400 x 600dpi
Laser no. 3 seller - Brother HLL2390DW: 2400 x 600dpi
Label no. 1 seller - Rollo Label Printer: 203dpi (symmetric)
Label no. 2 seller - DYMO LabelWriter 450: 600 x 300dpi

I have personally tested the DYMO LabelWriter and observe a significant difference in print quality when I print a 600x300dpi PDF vs a 300dpi PDF as exported by PIL.

Comment (tzjtan) [reply src/PIL/PdfImagePlugin.py]: The base branch didn't have a resolution sign check, but I have now added one in my update anyway.

**Comment by tzjtan** [comment]
Waited more than 2 months for merge but instead have to rebase. Rebase attempt is buggy, withdrawing contribution.

**Comment by radarhere** [comment]
This is being attempted again in PR #6961

**Pull request closed without merging**